components/libraryprovider.py

```diff
@@ -0,0 +1,26 @@
+#!/usr/bin/env python3
+
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import yaml
+
+from components.utilities import Struct
+from components.providerbase import BaseProvider, INeedsCommandProvider, INeedsLoggingProvider
+
+
+class LibraryProvider(BaseProvider, INeedsCommandProvider, INeedsLoggingProvider):
+    def __init__(self, config):
+        pass
+
+    def get_libraries(self, gecko_path):
+        libraries = []
+        mozilla_central_yamls = self.run(["find", gecko_path, "-type", "f", "-name", "moz.yaml"]).stdout.decode().strip().split("\n")
+
+        for file in mozilla_central_yamls:
+            with open(file, "r") as mozyaml:
+                new_library = yaml.safe_load(mozyaml.read())
+                new_library['yaml_path'] = file
+                libraries.append(Struct(**new_library))
```

Comment:
Instead of just sticking everything into a struct, I think we should have a validation function that reads the result of parsing the yaml and populates a library object that looks like (or identical) to the old one.

This will ensure that we're only accessing properties of the yaml file we care about, and we detect - at time of load - that all the info we have is available (as opposed to getting a 'key not found' exception in the middle of the execution.)

This will also enable us to run a task on autoland that asserts about the m-c repo and could cause backouts if we needed them - #107